### ISSUE TITLE
Error in read_tokens_ in 1.4.0 for a file that reads correctly in 1.3.1

### ISSUE THREAD

**Issue by martinmodrak** [tidyverse/readr#1140]
I am getting a cryptic message when reading a .csv file with readr 1.4.0, providing the `col_types` argument:

> Error in read_tokens_(data, tokenizer, col_specs, col_names, locale_, : Invalid input type, expected 'character' actual 'double'

The same file reads correctly:
- Without providing the `col_types` argument
- In readr 1.3.1 with the exact same call (including the `col_types` argument)

I cannot share the file as it contains patient data and don't have a better reprex, but I am sharing it here in hope others who encounter the error will find this and learn that they may work around it by installing 1.3.1 and hopefully they will be able to share their files. I understand that it is hard to take action on the issue without a reprex. Sorry if this is not appropriate and thanks for all the hard work on the package.

**Comment by jimhester** [comment]
You can use fake data to reproduce it, what you are passing as the col_types argument?

**Comment by sleyn** [comment]
Hello!

I had an issue with the same error.
The error was caused by `col_factor(levels = c(1, 2, 3, 4))`. It was fine with the previous version of readr (don't remember what I had before). With readr 1.4.0 I've got:
`Error in read_tokens_(data, tokenizer, col_specs, col_names, locale_, : Invalid input type, expected 'character' actual 'double'`
The error was solved with making the levels vector as a character vector `col_factor(levels = c('1', '2', '3', '4'))`.
